lib/linguist/generated.rb

```diff
@@ -708,6 +709,16 @@ def generated_jooq?
       lines.first(2).any? { |l| l.include? 'This file is generated by jOOQ.' }
     end
 
+    # Internal: Is this a generated Delphi Interface file for a type library?
+    #
+    # Delphi Type Library Import tool generates *_TLB.pas files based on .ridl files.
+    # They are not meant to be altered by humans.
+    #
+    # Returns true or false
+    def generated_pascal_tlb?
+      name =~ /_tlb\.pas$/i
```

Comment:
🤦 I completely missed this before, but this doesn't return a boolean which means it'll break the check for generated when the filename does _not_ match as it returns `nil` which is _not_ the same as `false`.

I'll open a PR now to correct this.

Reply:
https://github.com/github/linguist/pull/5034